### PR TITLE
TASK-10/setup-detox

### LINKED ISSUE #31: TASK-10: Setup Detox and Acceptance Tests
As a developer, I want to initialize Detox in order for the other developers to start testing their USs.

ACCEPTANCE CRITERIA:
* Detox is initialized in the repo
* Each current merged user story has a system acceptance test
* There exists a wiki page on how to use detox

IS BLOCKED BY:
* #27

### PR DESCRIPTION
**Pull request by RGPosadas**
* Before reviewing, **`please read wiki!`**
* Wiki page found [here](https://github.com/RGPosadas/WayFinder/wiki/HOW-TO:-Acceptance-Tests-with-Detox)
* For macOS users: you are the main reviewers
* For non-macOS users: here are the [GIFs](https://github.com/RGPosadas/WayFinder/wiki/Acceptance-Tests-GIFs)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -10,9 +10,8 @@ before_script:
   - curl https://pre-commit.com/install-local.py | python -
   - pre-commit install
 script:
-  - pre-commit run -a
   - npm install -g expo-cli --loglevel=error
   - npm install --log-level=error
   - echo "fakeSecret" >> secrets.js
-  - npm run lint
+  - npm run supercheck-travis
```
Comment (cristian-aldea) [inline .travis.yml:16]: nice name :3

File: README.md [hunk 2]
```diff
@@ -35,10 +35,32 @@ To run this react-native application and start development:
    - `npm install`
 4. Run project:
 
-   - `npm start`
+   - `expo r -c`
```
Comment (cristian-aldea) [inline README.md:47]: why change this?

File: package.json [hunk 3]
```diff
@@ -6,16 +6,22 @@
     "eject": "expo eject",
```
Comment (cristian-aldea) [inline package.json:8]: Make sure to run `npx sort-package-json` to have everything nice and sorted

**Review by cristian-aldea — COMMENTED** [review]
Looks good! Had a few minor, non-blocking questions.

For the gifs, why are they in a "markdown" folder?

File: .travis.yml (hunk 1 above)
Comment (RGPosadas) [reply .travis.yml:16]: thanks~

File: README.md (hunk 2 above)
Comment (RGPosadas) [reply README.md:47]: Had some cache issues with just plain `npm start`, and the `-c` fixed it for expo
... also `expo r` is much shorter than `npm start` lmao it was a nitpick

**Comment by RGPosadas** [comment]
@TheGreatMarkus 
For `assets/markdown`, they're used here:
https://github.com/RGPosadas/WayFinder/wiki/Acceptance-Tests-GIFs
It's used for markdown purposes, and I named it accordingly (unless there's a naming convention somewhere out there?)

File: package.json (hunk 3 above)
Comment (RGPosadas) [reply package.json:8]: done

File: README.md (hunk 2 above)
Comment (cristian-aldea) [reply README.md:47]: Yeah, I just meant clearing the cache prob isn't necessary for first time setup. its doesn't change anything either way.

**Comment by cristian-aldea** [comment]
> @TheGreatMarkus
> For `assets/markdown`, they're used here:
> https://github.com/RGPosadas/WayFinder/wiki/Acceptance-Tests-GIFs
> It's used for markdown purposes, and I named it accordingly (unless there's a naming convention somewhere out there?)

Oh interesting, I didn't know you were linking to the actual project content with the gifs in the wiki. I was just curious as for the `markdown` name, its all good there's aren't any conventions haha.

**Review by cristian-aldea — APPROVED** [review]
Addressed my questions, everything looks great! Unfortunately I can't run the tests myself, but the GIFs looks right :)

File: src/components/map-overlays/map-overlays.component.tsx [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import React, { useEffect, useState } from "react";
 import { Overlay } from "react-native-maps";
-import { StyleSheet } from "react-native";
+import { StyleSheet, View } from "react-native";
```
Comment (osahtout) [inline src/components/map-overlays/map-overlays.component.tsx:3]: If Imports are not being used, can they be removed?

File: src/screens/map-screen/map-screen.component.tsx [hunk 5]
```diff
@@ -1,5 +1,5 @@
 import React, { useState, useEffect, useRef } from "react";
-import { StyleSheet, View } from "react-native";
+import { StyleSheet, View, ScrollView } from "react-native";
```
Comment (osahtout) [inline src/screens/map-screen/map-screen.component.tsx:2]: Same here

**Review by osahtout — APPROVED** [review]

**Review by cindyslittleplanet — APPROVED** [review]
Tested on Macbook Air, all tests passes. 
It looks good!

**Pull request merged**